### PR TITLE
✨ Bump Gophercloud to v2 beta

### PR DESCRIPTION
**Pull request by EmilienM**
**What this PR does / why we need it**:
Update Gophercloud + Utils to v2-beta

/hold

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,23 @@
 module sigs.k8s.io/cluster-api-provider-openstack
 
-go 1.21
+go 1.21.6
```
Comment (pierreprinetti) [inline go.mod:3]: v1.22.0 is out by the way

Comment (EmilienM) [reply go.mod:3]: ok, I think it'll be nice to do it via another PR though; just in case

File: go.mod [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 require (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/go-logr/logr v1.3.0
 	github.com/golang/mock v1.6.0
 	github.com/google/go-cmp v0.6.0
 	github.com/google/gofuzz v1.2.0
-	github.com/gophercloud/gophercloud v1.7.0
-	github.com/gophercloud/utils v0.0.0-20231010081019-80377eca5d56
+	github.com/gophercloud/gophercloud/v2 v2.0.0-beta.1
+	github.com/gophercloud/utils/v2 v2.0.0-20240209190212-277d485b2674
```
Comment (EmilienM) [inline go.mod:14]: this is not master/head, but a `v2-beta` branch.

File: go.mod (hunk 1 above)
Comment (EmilienM) [reply go.mod:3]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1877

**Comment by EmilienM** [comment]
This is ready for review.

**Review by lentzi90 — COMMENTED** [review]
Awesome!
/approve

**Comment by huxcrux** [comment]
Sweet to get all new fixes :)
/lgtm

**Comment by EmilienM** [comment]
I want @mdbooth's opinion on this one. He verbally agreed to do it but he might have another opinion today. Please hold cancel if you're fine with it.

**Review by mandre — COMMENTED** [review]
Thanks for doing this @EmilienM. The gophercloud v2 changes are (for now) minimal compared to v1 and I don't expect any breakage.

/lgtm

**Comment by pierreprinetti** [comment]
Please note that we are likely to change all function signatures in `v2-beta.2` to include a `context.Context`.

**Comment by mdbooth** [comment]
> Please note that we are likely to change all function signatures in `v2-beta.2` to include a `context.Context`.

Very much in favour of landing this once v2 lands, but not before, for example because of the possibility of changes like the above prior to release.

I think gophercloud is pretty close to releasing that though, right?

**Comment by jichenjc** [comment]
>>>Update Gophercloud + Utils to v2-beta
>>>I think gophercloud is pretty close to releasing that though, right?

also curious about when v2 will land ,something like CPO will benefit as well if some enhancement can be adopted

**Comment by pierreprinetti** [comment]
>I think gophercloud is pretty close to releasing that though, right?

Yes although I'd measure the ETA in weeks rather than days or hours.

**Comment by EmilienM** [comment]
@pierreprinetti sounds like we could update this one to use beta-2 but this will need a bunch of changes for ctx, right?

**Comment by pierreprinetti** [comment]
> @pierreprinetti sounds like we could update this one to use beta-2 but this will need a bunch of changes for ctx, right?

correct beta.2 adds ctx everywhere

**Comment by pierreprinetti** [comment]
Updated to v2.0.0-beta.2. Calls to Gophercloud now use `context.TODO()`. For now I am deferring the decision whether to actually use the context in the whole CAPO.

**Comment by EmilienM** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2107

**Pull request closed without merging**